core/shared/src/main/scala/zio/stm/TMap.scala

```diff
@@ -267,6 +276,12 @@ final class TMap[K, V] private (
       TExit.unit
     })
 
+  /**
+   * Retains bindings matching predicate and returns removed bindings.
+   */
+  def retainIfReturning(p: (K, V) => Boolean): USTM[Chunk[(K, V)]] =
+    removeIfReturning(!p(_, _))
```

Comment:
Sorry for being nitpicky, but this is a bit odd as we have a bit convoluted situation:
- `retainIfReturning` is implemented in terms of `removeIfReturning`
- `removeIf` is implemented in terms of `retainIf`